### PR TITLE
Validate inputs and handle errors

### PR DESCRIPTION
**Pull request by ktp242**
This PR focus on adding the functions to validation the date inputs on the client side. Also, the app now displays the errors if there is any from the input.

If either of the date format is not as `MM/DD/YYYY`, include empty value, after hitting the submit button, the instruction text will change to indicate a further info for patron to enter the correct format of the date. And the request will not be submitted. The result thus won't be changed.

However, if something wrong happened from the server and thus the app doesn't get any valid result back, the result will indicate the error, as `The input dates have invalid format or value, please check again.`

### CODE REVIEW

File: src/shared/containers/Refile/RefileErrorsForm.jsx [hunk 1]
```diff
@@ -50,7 +49,11 @@ class RefileErrorsForm extends Component {
     const currentState = state;
 
     if (typeof currentState.fieldErrors === 'object' && isEmpty(currentState.fieldErrors[field])) {
-      currentState.fieldErrors[field] = errorString;
+      const currentFieldErrorsState = assign(currentState.fieldErrors, { [field]: errorString });
```
Comment (thisisstephenbetts) [inline src/shared/containers/Refile/RefileErrorsForm.jsx]: Matter of interest, could we do this with the following? Again, not a big deal :-)

```
const currentFieldErrorsState = {...currentState.fieldErrors, [field]: errorString };
```

File: src/shared/containers/Refile/RefileErrorsForm.jsx [hunk 2]
```diff
@@ -105,31 +96,58 @@ class RefileErrorsForm extends Component {
...
+    // Checks if it has a valid date format. The Regex check if the inputs are digits
+    // and if they have right number of digits
+    const dateMatches = date.match(/^(\d{2})\/(\d{2})\/(?:\d{4})$/);
+
+    if (!dateMatches) {
+      return false;
+    }
+
+    // Checks if the month is valid
+    if (parseInt(dateArray[0], 10) < 1 || parseInt(dateArray[0], 10) > 12) {
```
Comment (thisisstephenbetts) [inline src/shared/containers/Refile/RefileErrorsForm.jsx]: Very, very minor point, makes no difference, but I'd do 

```
let month = parseInt(dateArray[0], 10)
```

More readable, don't have to repeat the `parseInt` with its base-10, and only one thing to change if the regex changes. Obviously makes no practical difference here, but I prefer it for general style.

File: src/shared/containers/Refile/RefileErrorsForm.jsx [hunk 3]
```diff
@@ -185,19 +204,28 @@ class RefileErrorsForm extends Component {
...
-    //   this.validateField(key, this.state, true);
-    // });
-
-    if (isEmpty(this.state.fieldErrors)) {
+  handleFormSubmit(resetOffset) {
+    // The array that stores the results after validating the date inputs
+    const fieldErrorsArray = [];
+
+    // Iterate through date fields and ensure all fields are valid
+    forIn(this.state.formFields, (value, key) => {
```
Comment (thisisstephenbetts) [inline src/shared/containers/Refile/RefileErrorsForm.jsx:220]: Could do this with a `forEach`?

(Don't get me wrong, I don't hate lo-dash, just not sure it's so needed anymore — but very happy to discuss, and happy to just leave as it is for now)

**Review by thisisstephenbetts — APPROVED** [review]
Very minor suggestions/discussion points. No big deal. Looks good!

File: src/shared/containers/Refile/RefileErrorsForm.jsx (hunk 1 above)
Comment (ktp242) [reply src/shared/containers/Refile/RefileErrorsForm.jsx]: We can! Thanks for telling me this. I just tested it. Will update it.

File: src/shared/containers/Refile/RefileErrorsForm.jsx (hunk 2 above)
Comment (ktp242) [reply src/shared/containers/Refile/RefileErrorsForm.jsx]: I can update it! Good suggestions!

File: src/shared/containers/Refile/RefileErrorsForm.jsx (hunk 3 above)
Comment (ktp242) [reply src/shared/containers/Refile/RefileErrorsForm.jsx:220]: Maybe we could! I could try testing it.

Comment (ktp242) [reply src/shared/containers/Refile/RefileErrorsForm.jsx:220]: ahhh you know what? `this.state.formFields` is not an array but an object, so we can't use `forEach` with it....

Comment (thisisstephenbetts) [reply src/shared/containers/Refile/RefileErrorsForm.jsx:220]: Ah, my bad! Thanks for checking.

**Pull request merged**